plugins/lang/python/python_loader.cc

```diff
@@ -567,11 +570,28 @@ class mldb_wrapper(object):
                                                async=True)
             self.create_dataset = self._mldb.create_dataset
 
+        def _follow_redirect(self, url, counter):
+            # somewhat copy pasted from _perform, but gives a nicer stacktrace
+            # on failure
+            if counter == 0:
+                raise mldb_wrapper.TooManyRedirectsException()
+
+            raw_res = self._mldb.perform('GET', url)
+            response = mldb_wrapper.Response(url, raw_res)
+            if response.status_code < 200 or response.status_code >= 400:
+                raise mldb_wrapper.ResponseException(response)
+            if response.status_code >= 300:
+                return self._follow_redirect(response.headers['location'],
+                                             counter - 1)
+            return response
+
         def _perform(self, method, url, *args, **kwargs):
             raw_res = self._mldb.perform(method, url, *args, **kwargs)
             response = mldb_wrapper.Response(url, raw_res)
             if response.status_code < 200 or response.status_code >= 400:
                 raise mldb_wrapper.ResponseException(response)
+            if response.status_code >= 300:
+                return self._follow_redirect(response.headers['location'], 10)
```

Comment:
This is here because the auto redirect portion of the python wrapper wasn't done yet and the attached test covers it.